server/src/test/java/org/elasticsearch/action/bulk/BulkIntegrationIT.java

```diff
@@ -156,4 +165,39 @@ private void createSamplePipeline(String pipelineId) throws IOException, Executi
 
         assertTrue(acknowledgedResponse.isAcknowledged());
     }
+
+    /** This test ensures that index deletion makes indexing fail quickly, not wait on the index that has disappeared */
+    public void testDeleteIndexWhileIndexing() throws Exception {
+        String index = "deleted_while_indexing";
+        createIndex(index);
+        AtomicBoolean stopped = new AtomicBoolean();
+        Thread[] threads = new Thread[between(1, 4)];
+        AtomicInteger docID = new AtomicInteger();
+        for (int i = 0; i < threads.length; i++) {
+            threads[i] = new Thread(() -> {
+                while (stopped.get() == false && docID.get() < 5000) {
+                    String id = Integer.toString(docID.incrementAndGet());
+                    try {
+                        IndexResponse response = client().prepareIndex(index).setId(id)
+                            .setSource(Map.of("f" + randomIntBetween(1, 10), randomNonNegativeLong()), XContentType.JSON).get();
+                        assertThat(response.getResult(), isOneOf(CREATED, UPDATED));
+                        logger.info("--> index id={} seq_no={}", response.getId(), response.getSeqNo());
+                    } catch (ElasticsearchException ignore) {
+                        logger.info("--> fail to index id={}", id);
+                    }
+                }
+            });
+            threads[i].start();
+        }
+        ensureGreen(index);
+        assertBusy(() -> assertThat(docID.get(), greaterThanOrEqualTo(1)));
+        assertAcked(client().admin().indices().prepareDelete(index));
+        stopped.set(true);
+        for (Thread thread : threads) {
+            thread.join(10000); // only wait 10 seconds here
```

Comment:
Maybe just make it `30s` ... 10s for this kind of thing alway seems like it could randomly fail, maybe not as likely anymore but why take the risk? Also, maybe make it relative to `org.elasticsearch.action.support.replication.ReplicationRequest#DEFAULT_TIMEOUT` to make the intent of the test clearer clearer?